tests/e2e/test_pv_scale_ocs_create_delete_pvcs.py

```diff
@@ -0,0 +1,338 @@
+            #         executor.submit(helpers.create_pod,
+            #                         interface_type=constants.CEPHBLOCKPOOL,
+            #                         pvc_name=pvc_obj.name, do_reload=False,
+            #                         namespace=self.namespace))
+            #     log.info(f"Created pod {pod_name} using pvc {pvc_obj.name}")
+
+        temp_pod_objs = [pod_obj.result() for pod_obj in future_temp_pod_objs]
+        log.info(f"Create {len(temp_pod_objs)} pods")
+        self.all_pod_obj.extend(temp_pod_objs)
+        # Check for all the pods are in Running state, in above pod creation
+        # Not waiting for the pod to be created because of threads usage
+        with ThreadPoolExecutor() as executor:
+            for pod in temp_pod_objs:
+                future_pod_objs.append(
+                    executor.submit(helpers.wait_for_resource_state,
+                                    pod, 'Running', timeout=self.timeout))
+        check_objs = [obj.result() for obj in future_pod_objs]
+        # If pods not up raising exception/failure
+        if False in check_objs:
+            raise TimeoutExpiredError
+        # Added run IO part for TC 'OCS-679'
+        if self.test_case == 'ocs679':
+            with ThreadPoolExecutor() as executor:
+                for pod_obj in temp_pod_objs:
+                    executor.submit(pod_obj.run_io('fs', size='1G'))
+
+        log.info(f"all_pod_list {self.all_pod_obj}")
+        log.info(f"all_pvc_list {self.all_pvc_obj}")
+        log.info(f"delete_pod_count {self.delete_pod_count}")
+
+    def delete_pvc_pod(self):
+        """
+        def to delete pvc and pod based on the delete pod count.
+        """
+        log.info(f"Delete {self.delete_pod_count} pods and respective pvcs")
+        temp_pod_list = random.choices(self.all_pod_obj, k=self.delete_pod_count)
+        temp_pvc_list = []
+        for pod_obj in temp_pod_list:
+            for pvc_obj in self.all_pvc_obj:
+                if pod_obj.pvcname == pvc_obj.name:
+                    temp_pvc_list.append(pvc_obj)
+                    log.info(f"printing the pvc to be deleted {pvc_obj.name}")
+                    log.info(f"Print pvc obj {pvc_obj}")
+                    log.info(f"Print all_pvc_obj {self.all_pvc_obj}")
+                    self.all_pvc_obj.remove(pvc_obj)
+            log.info(f"Printing the pod to be deleted {pod_obj}")
+            if pod_obj in self.all_pod_obj:
+                self.all_pod_obj.remove(pod_obj)
+        helpers.delete_objs_parallel(temp_pod_list)
+        helpers.delete_objs_parallel(temp_pvc_list)
+
+    def cleanup(self):
+        """
+        def to cleanup the SC, PVC and POD objects parallel.
+        """
+        helpers.delete_objs_parallel(self.all_pod_obj)
+        helpers.delete_objs_parallel(self.all_pvc_obj)
+        self.rbd_sc_obj.delete()
+        self.cephfs_sc_obj.delete()
+
+
+@scale
+@pytest.mark.usefixtures(
+    create_project.__name__,
```

Comment:
consider using factory fixtures

Reply:
will sync up with you guys to resolve this.

Reply:
@ebenahar Implemented factory fixtures, would request to review the changes.